File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x<0:
            return self.reverse(-x)*-1
        ans = int(str(x)[::-1])
        maxi = (1<<31)-1
        return ans if (-maxi)-1<=ans<=maxi else 0
        

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long ans=0;
        while(x != 0){
            ans = (ans*10) + (x%10);
            x/=10;
        }
        return (ans<=Integer.MAX_VALUE && ans>=Integer.MIN_VALUE) ? (int)ans : 0;
    }
}